blaze/server/tests/test_server.py

```diff
@@ -386,6 +386,44 @@ def test_server_accepts_non_nonzero_ables():
 
 
 @pytest.mark.parametrize('serial', all_formats)
+def test_map_client_server(iris_server, serial):
+    print(serial)
+    test = iris_server
+    t = symbol('t', discover(iris))
+    expr = t.species.map(len, 'int')
+    query = {'expr': to_tree(expr)}
+    response = test.post('/compute',
+                         data=serial.dumps(query),
+                         headers=mimetype(serial))
+    assert 'OK' in response.status
+    respdata = serial.loads(response.data)
+    result = serial.data_loads(respdata['data'])
+
+    istrue = result == compute(expr, {t: iris}, return_type=list)
+    try:
+        assert istrue
+    except:
```

Comment:
The fastmsgpack was added to dramatically speed up the performance of the blaze server. The bottleneck in our production blaze servers and clients was actually serializing and deserializing the data. The general problem was the the dataframe was getting serialized as a list of records and then when rehydrating the data we would call object_hook tens of millions of times. object_hook was consistently in the top 5 ncalls when profiling, so even with a percall that rounded to 0s, it was a large part of the runtime. The fastmsgpack format is designed to send each block of the dataframe as a single buffer which is then compressed with blosc. This is why we have a seperate `data_(dumps|loads)` and `dumps|loads`. Basically, the data field of the response record is actually a seperately serialized dataframe which has been compressed. This allows us to send a smaller response which is then deserialized much more quickly.

There is also the potential to reduce the copying, If we deserialize as a list, then we need to take the bytes off the wire and turn them into python integers which is going to trigger thousands of allocations. This is really painful because we often just want to push that right back into an ndarray. I backported some code I added to pandas to prevent the copy and still give us a mutable buffer, which can be found here: https://github.com/blaze/blaze/blob/master/blaze/server/serialization.py#L117 (note, I actually just found a bug there, I will push a fix now). In the blaze version, we only slightly break the abstraction that bytes are immutable; however, the pandas version does this in a much more sane way: https://github.com/pydata/pandas/blob/master/pandas/io/packers.py#L325.